src/Storageless/Session/ArrayAccessor.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+declare(strict_types = 1);
+
+namespace PSR7Sessions\Storageless\Session;
+
+/**
+ * Class ArrayAccessor
+ *
+ * Wrapper class around SessionInterface with ArrayAccess Interface
+ */
+class ArrayAccessor implements \ArrayAccess
+{
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * ArrayAccessor constructor.
+     *
+     * @param SessionInterface $session
+     */
+    public function __construct(SessionInterface $session)
+    {
+        $this->session = $session;
+    }
+
+    /**
+     * Proxy function to SessionInterface
+     *
+     * @param string $method
+     * @param array  $args
+     *
+     * @return mixed
+     */
+    public function __call($method, $args)
```

Comment:
Why is this here? Magic functions are hard to grasp for the developer? This looks like you attempt to implement the original SessionInterface without explicitly implementing it. I'd consider this proxy useless for anyone using proper IDEs with code completion because this accessor object lacks the `implements SessionInterface` part. On the other hand, adding that interface would force you to implement all the methods explicitly, adding boilerplate code in every one of them.